cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -114,10 +116,14 @@ void SparsePoseGraph::AddScan(
     submap_states_.back().trajectory = submaps;
     CHECK_EQ(submap_states_.size(), submap_indices_.size());
   }
+  // ... One call before the first of insertion_submaps was finished
   const mapping::Submap* const finished_submap =
       insertion_submaps.front()->finished_probability_grid != nullptr
           ? insertion_submaps.front()
           : nullptr;
+  // by providing finished_submap (!=nullptr) to ComputeConstraintsForScan
```

Comment:
Seems this information should be moved to the comment for `ComputeConstraintsForScan()` in the header?